src/material/dialog/dialog-container.ts

```diff
@@ -218,7 +215,13 @@ export abstract class _MatDialogContainerBase extends BasePortalOutlet {
         break;
       case true:
       case 'first-tabbable':
-        this._focusTrap.focusInitialElementWhenReady();
+        this._focusTrap.focusInitialElementWhenReady().then(focusedSuccessfully => {
```

Comment:
Im not too sure if this can have a target of patch. This is touching code from #22780, which will go into the next minor release.

Reply:
Also, does sidenav or bottom sheet need special handling too?

Reply:
@amysorto Special handling when opening to support screen readers? Probably not, unless they did the bouncing of focus that dialog did prior to this PR.

Reply:
- [ ] Check that bottom sheet behaves properly in a screen reader when opened
- [ ] Check that sidenav behaves properly in a screen reader when opened

Reply:
@zarend found some similar issues with Bottom Sheet and he's investigating them now.